core/src/main/scala/org/apache/spark/util/HadoopFSUtils.scala

```diff
@@ -308,4 +340,14 @@ private[spark] object HadoopFSUtils extends Logging {
     val include = pathName.startsWith("_common_metadata") || pathName.startsWith("_metadata")
     exclude && !include
   }
+
+  /** Checks if we should filter out this path. */
+  def shouldFilterOutPath(path: String): Boolean = {
+    val exclude = (path.contains("/_") && !path.contains("=")) || path.contains("/.")
+    val include = path.contains("/_common_metadata/") ||
+      path.endsWith("/_common_metadata") ||
+      path.contains("/_metadata/") ||
+      path.endsWith("/_metadata")
+    (exclude && !include) || shouldFilterOutPathName(path)
```

Comment:
Isn't `shouldFilterOutPathName` already covering these patterns?

Reply:
- `shouldFilterOutPathName` only handles `file name`. 
- `shouldFilterOutPath` handles the intermediate directory in the path like `s3://abc/_metadata/1.txt`.